trino/trino.go

```diff
@@ -1215,22 +1223,34 @@ func (qr *driverRows) Next(dest []driver.Value) error {
 			return err
 		}
 	}
+
 	if len(qr.coltype) == 0 {
 		qr.err = sql.ErrNoRows
 		return qr.err
 	}
-	for i, v := range qr.coltype {
-		if i > len(dest)-1 {
+
+	row := qr.data[qr.rowindex]
+	for i, colType := range qr.coltype {
+		if i >= len(dest) {
 			break
 		}
-		vv, err := v.ConvertValue(qr.data[qr.rowindex][i])
+		val, err := colType.ConvertValue(row[i])
 		if err != nil {
 			qr.err = err
 			return err
 		}
-		dest[i] = vv
+		dest[i] = val
 	}
+
 	qr.rowindex++
+
+	// Prefetch next set of rows
```

Comment:
Useless comment - it explains the what, which is pretty obvious, but not the why, which is not. Why do we need to do this? If the user doesn't request any more data, we'll waste resources on one round-trip.

Can we do this only when cancelling the query?